### PR TITLE
Numdiff and Makefile mods

### PR DESCRIPTION
**Pull request by reubendb**
This PR contains python script `numdiff.py` to compare files numerically for regression testing and other minor modifications in Makefile and build-related scripts for Poplar compilers.

### CODE REVIEW

File: builds/scripts/amake-paris-pfft.sh [hunk 1]
```diff
@@ -1,4 +1,5 @@
 #!/bin/bash
+module restore PrgEnv-cray
```
Comment (twhite-cray) [inline builds/scripts/amake-paris-pfft.sh:2]: Did you try this change? The `pfft` module already does a `module restore PrgEnv-cray`. And I think doing `module restore PrgEnv-cray` first will wipe out the `module use` that allows you to see the `pfft` module.

Comment (reubendb) [reply builds/scripts/amake-paris-pfft.sh:2]: For some reason I can't see `pfft` module either way, so with / without the `module restore PrgEnv-cray` this script fails to build. Is there a missing `module use ...` that is supposed to be there?

Comment (twhite-cray) [reply builds/scripts/amake-paris-pfft.sh:2]: Yeah, sorry, it's in my default environment.
```
module use /home/users/twhite/share/modulefiles
```

Comment (reubendb) [reply builds/scripts/amake-paris-pfft.sh]: OK, thanks. I think e2533b0 fixes this. The scripts now build correctly for me.

Comment (reubendb) [reply builds/scripts/amake-paris-pfft.sh]: I am not sure why `module restore` wipes modulepath added by `module use`; I don't think I'd agree with that behavior but I guess that's a side point ...

**Pull request merged**